src/commands/branch.rs

```diff
@@ -293,15 +294,34 @@ fn cmd_branch_list(
 
     let mut formatter = ui.stdout_formatter();
     let formatter = formatter.as_mut();
-    for (name, branch_target) in repo.view().branches() {
+
+    let mut all_branches = repo.view().branches().clone();
+    for (branch_name, git_tracking_target) in git_tracking_branches(repo.view()) {
+        let branch_target = all_branches.entry(branch_name.to_owned()).or_default();
+        if branch_target.remote_targets.contains_key("git") {
+            // TODO(#1690): There should be a mechanism to prevent importing a
+            // remote named "git" in `jj git import`.
+            // TODO: This is not currently tested
+            writeln!(
+                formatter.labeled("warning"),
```

Comment:
Should we use `ui.warning()` here? Then the output will be to stderr, which is probably good.

Reply:
Certainly! This also allows me to rearrange the code in a bit.